### PR TITLE
feat: Metric engine skeleton

### PR DESCRIPTION
**Pull request by waynexia**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

Define a basic skeleton for metric engine

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.

## Refer to a related PR or issue link (optional)
part of https://github.com/GreptimeTeam/greptimedb/issues/2070

### CODE REVIEW

File: src/metric-engine/src/engine.rs [hunk 1]
```diff
@@ -0,0 +1,317 @@
...
+        // create the region
+        engine
+            .handle_request(region_id, RegionRequest::Create(region_create_request))
+            .await
+            .unwrap();
+
+        let mut ls_result = tokio::fs::read_dir(&engine_dir).await.unwrap();
+        while let Some(dir) = ls_result.next_entry().await.unwrap() {
+            info!("[DEBUG] {dir:?}");
+        }
```
Comment (evenyag) [inline src/metric-engine/src/engine.rs:306]:
```suggestion
```

File: src/metric-engine/src/engine.rs [hunk 2]
```diff
@@ -0,0 +1,317 @@
...
+    async fn create_metadata_region() {
+        common_telemetry::init_default_ut_logging();
+
+        let env = TestEnv::new().await;
+        let mito = env.mito();
+        let engine = MetricEngine {
+            inner: Arc::new(MetricEngineInner { mito }),
+        };
+        let engine_dir = env.data_home();
+        info!("[DEBUG] {engine_dir}");
```
Comment (evenyag) [inline src/metric-engine/src/engine.rs:285]:
```suggestion
```

File: src/mito2/src/engine.rs [hunk 3]
```diff
@@ -28,7 +28,8 @@ mod create_test;
 mod drop_test;
 #[cfg(test)]
 mod flush_test;
-#[cfg(test)]
+#[cfg(any(test, feature = "test"))]
+#[cfg_attr(feature = "test", allow(unused))]
 pub(crate) mod listener;
```
Comment (evenyag) [inline src/mito2/src/engine.rs:33]: Do we still need `allow(unused)` if we pub this mod? It's fine to pub it under the test feature.

**Review by evenyag — APPROVED** [review]

**Review by fengjiachun — APPROVED** [review]
LGTM

**Pull request merged**